instance/models/load_balancer.py

```diff
@@ -111,6 +148,27 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.domain
 
+    def modifies_configuration(method):  # nopep8 # pylint: disable=no-self-argument
+        """
+        Flag an internal method as one that modifies the state of load balancer configuration,
+        to allow managing configuration state outside and around the method call.
+        """
+        @functools.wraps(method)
+        def wrapper(self, *args, **kwargs):
+            """
+            Used to wrap the function that modified load balancer configuration state.
+            """
+            self.configuration_version += 1
```

Comment:
This increment is subject to a race condition - does that matter?

If necessary it can be changed to use an [F expression](https://docs.djangoproject.com/en/1.11/ref/models/expressions/#f-expressions):

```python
from django.db.models import F
...
self.configuration_version = F('configuration_version') + 1
self.save()
self.refresh_from_db()  # Required to read the result, and to avoid future .save() calls from incrementing the value again
```

Reply:
@bradenmacdonald We've discussed this, and came to the conclusion that it doesn't matter.  If tow processes increase the version number concurrently, we might only increase by one instead of two, but that would still mark the LB as dirty, and only one of the processes will be able to grab the lock to actually reconfigure it.  We should add a comment here, though, to explain why this is fine.

Reply:
Will explain this, yep.